mmv1/templates/terraform/pre_update/netapp_storagepool.go.erb

```diff
@@ -0,0 +1,65 @@
+// detect manual zone switches for service level FLEX
+
+if d.Get("service_level").(string) == "FLEX" {
+	// Check if this is zonal or regional Flex. Only continue for regional pool
+	_, hasZone := d.GetOk("zone")
+	_, hasReplicaZone := d.GetOk("replica_zone")
+	if hasZone && hasReplicaZone {
+		// For a zone switch, user needs to swap zone and replica_zone. Other changes are not allowed
+		if d.HasChange("zone") && d.HasChange("replica_zone") {
+			oldZone, newZone := d.GetChange("zone")
+			oldReplicaZone, newReplicaZone := d.GetChange("replica_zone")
+			if newZone == oldReplicaZone && newReplicaZone == oldZone {
+				rawurl, err := tpgresource.ReplaceVars(d, config, "{{NetappBasePath}}projects/{{project}}/locations/{{location}}/storagePools/{{name}}:switch")
+				if err != nil {
+					return err
+				}
+
+				reso, err := transport_tpg.SendRequest(transport_tpg.SendRequestOptions{
+					Config:    config,
+					Method:    "POST",
+					Project:   billingProject,
+					RawURL:    rawurl,
+					UserAgent: userAgent,
+					Timeout:   d.Timeout(schema.TimeoutUpdate),
+				})
+				if err != nil {
+					return fmt.Errorf("Error switching active zone for pool: %s, %v", d.Id(), err)
+				}
+
+				err = NetappOperationWaitTime(
+					config, reso, project, "Switching active pool zone", userAgent,
+					d.Timeout(schema.TimeoutDelete))
```

Comment:
@roop2 please change it to d.Timeout(schema.TimeoutUpdate))

Reply:
done